app.ts

```diff
@@ -0,0 +1,147 @@
+﻿// Onboarding Project - 2018
+
+const RECORD_COUNT: number = JSON.parse(getData("http://localhost:2050/recordCount")); // total number of records available in webserver
+let LAST_ENTRY, CURRENT_ENTRY; // entry numbers required for prev and next funtionality
+let resizeTimer; // delay required to reduce the number of calls made by browser when resizing the window
+let CONTROLS_HEIGHT;
+
+// makes api call and returns data
+function getData(dataLink: string) {
+    const apiCall = new XMLHttpRequest();
+    apiCall.open("GET", dataLink, false);
```

Comment:
Good idea to change this to use async AJAX call to get some exercise at working with callbacks (since you'll many of them in your time with JavaScript :)